### ISSUE TITLE
Autorecon1 Crashing #286  on fmriprep:22.0.0rc1  

### ISSUE THREAD

**Issue by juansanchezpena** [nipreps/fmriprep#2801]
### What happened?

nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node autorecon1.
No func data created. Saw the bug #286 but the problem persists.

### What command did you use?

```shell
I used docker swarm and also tried with docker
docker run -ti --rm \
-v /MRI_DATA/nyspi/patensasc/rawdata:/input:ro \
-v /MRI_DATA/nyspi/patensasc/derivatives:/output \
-v /MRI_DATA/freesurferlicense.txt:/opt/freesurfer/license.txt:ro \
nipreps/fmriprep:22.0.0rc1 \
/data /out \
participant --participant-label patensasc0085 --output-spaces T1w MNI152NLin2009cAsym:res-2 --cifti-output --output-layout legacy > /MRI_DATA/nyspi/patensasc/derivatives/fmriprep/logs/patensasc85.log 2>&1
```

### What version of fMRIPrep are you running?

22.0.0rc1

### How are you running fMRIPrep?

Docker

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
[juan@doctor 20220622-194930_b6c9a7c4-f8c0-42c1-9d24-c2bb21870bee]$ cayt crash-20220622-202601-root-autorecon1-280d52fd-362d-4982-983c-78bdc47b4a87.txt 
cayt: Command not found.
[juan@doctor 20220622-194930_b6c9a7c4-f8c0-42c1-9d24-c2bb21870bee]$ cat crash-20220622-202601-root-autorecon1-280d52fd-362d-4982-983c-78bdc47b4a87.txt 
Node: fmriprep_22_0_wf.single_subject_patensasc0085_wf.anat_preproc_wf.surface_recon_wf.autorecon1
Working directory: /tmp/work/fmriprep_22_0_wf/single_subject_patensasc0085_wf/anat_preproc_wf/surface_recon_wf/autorecon1

Node inputs:

FLAIR_file = <undefined>
T1_files = <undefined>
T2_file = <undefined>
args = <undefined>
big_ventricles = <undefined>
brainstem = <undefined>
directive = autorecon1
environ = {}
expert = <undefined>
flags = <undefined>
hemi = <undefined>
hippocampal_subfields_T1 = <undefined>
hippocampal_subfields_T2 = <undefined>
hires = <undefined>
mprage = <undefined>
mri_aparc2aseg = <undefined>
mri_ca_label = <undefined>
mri_ca_normalize = <undefined>
mri_ca_register = <undefined>
mri_edit_wm_with_aseg = <undefined>
mri_em_register = <undefined>
mri_fill = <undefined>
mri_mask = <undefined>
mri_normalize = <undefined>
mri_pretess = <undefined>
mri_remove_neck = <undefined>
mri_segment = <undefined>
mri_segstats = <undefined>
mri_tessellate = <undefined>
mri_watershed = <undefined>
mris_anatomical_stats = <undefined>
mris_ca_label = <undefined>
mris_fix_topology = <undefined>
mris_inflate = <undefined>
mris_make_surfaces = <undefined>
mris_register = <undefined>
mris_smooth = <undefined>
mris_sphere = <undefined>
mris_surf2vol = <undefined>
mrisp_paint = <undefined>
openmp = 8
parallel = <undefined>
steps = <undefined>
subject_id = recon_all
subjects_dir = <undefined>
talairach = <undefined>
use_FLAIR = <undefined>
use_T2 = <undefined>
xopts = <undefined>

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 524, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 642, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 750, in _run_command
    raise NodeExecutionError(
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node autorecon1.

RuntimeError: subprocess exited with code 1.
```

### Additional information / screenshots

]

**Comment by mgxd** [comment]
This is related to https://github.com/nipreps/smriprep/issues/286 - I thought the problem was fixed in 22.1.0rc1, but perhaps not.

To help debug this, can you follow the recommendation in https://github.com/nipreps/smriprep/issues/286#issuecomment-1146022969? That will help us get a better picture of what's going wrong.

**Comment by smeisler** [comment]
Hi, I opened the original smriprep issue. I just learned tha `getfullpath` from FS is also not imported and may be called by recon-all, leading to an error.

**Comment by mgxd** [comment]
We released `22.0.0rc2` - hopefully that shuts the door on this autorecon1 saga.

**Comment by juansanchezpena** [comment]
Awesome will test

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by smeisler** [comment]
@juansanchezpena can you follow the debugging instructions on the smriprep issues page (https://github.com/nipreps/smriprep/issues/286)? That is, go into the singularity shell and run the command from there?

**Comment by juansanchezpena** [comment]
I run docker but tried without swarms and got the same error:


Node: fmriprep_22_0_wf.single_subject_patensasc0085_wf.anat_preproc_wf.surface_recon_wf.autorecon1
Working directory: /workdir/work/fmriprep_22_0_wf/single_subject_patensasc0085_wf/anat_preproc_wf/surface_recon_wf/autorecon1

Node inputs:

FLAIR_file = <undefined>
T1_files = ['/workdir/work/fmriprep_22_0_wf/single_subject_patensasc0085_wf/anat_preproc_wf/anat_template_wf/t1w_merge/sub-patensasc0085_ses-1173_run-01_T1w_template.nii.gz']
T2_file = /input/sub-patensasc0085/ses-1173/anat/sub-patensasc0085_ses-1173_run-01_T2w.nii.gz
args = <undefined>
big_ventricles = <undefined>
brainstem = <undefined>
directive = autorecon1
environ = {}
expert = <undefined>
flags = ['-noskullstrip']
hemi = <undefined>
hippocampal_subfields_T1 = <undefined>
hippocampal_subfields_T2 = <undefined>
hires = True
mprage = <undefined>
mri_aparc2aseg = <undefined>
mri_ca_label = <undefined>
mri_ca_normalize = <undefined>
mri_ca_register = <undefined>
mri_edit_wm_with_aseg = <undefined>
mri_em_register = <undefined>
mri_fill = <undefined>
mri_mask = <undefined>
mri_normalize = <undefined>
mri_pretess = <undefined>
mri_remove_neck = <undefined>
mri_segment = <undefined>
mri_segstats = <undefined>
mri_tessellate = <undefined>
mri_watershed = <undefined>
mris_anatomical_stats = <undefined>
mris_ca_label = <undefined>
mris_fix_topology = <undefined>
mris_inflate = -n 50
mris_make_surfaces = <undefined>
mris_register = <undefined>
mris_smooth = <undefined>
mris_sphere = <undefined>
mris_surf2vol = <undefined>
mrisp_paint = <undefined>
openmp = 8
parallel = <undefined>
steps = <undefined>
subject_id = sub-patensasc0085
subjects_dir = /output/freesurfer
talairach = <undefined>
use_FLAIR = <undefined>
use_T2 = <undefined>
xopts = <undefined>

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 524, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 642, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 750, in _run_command
    raise NodeExecutionError(
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node autorecon1.

RuntimeError: subprocess exited with code 1.

**Comment by juansanchezpena** [comment]
There are no scripts outputs, When I log into the container I dont see the freesurfer bin

 
![Screen Shot 2022-07-11 at 3 28 52 PM](https://user-images.githubusercontent.com/72615316/178343280-da2fea0b-1d8c-47b5-a62e-41acfb85ffe3.png)

**Comment by effigies** [comment]
Apologies, I meant to type `recon-all`. I've updated the above command.

**Comment by effigies** [comment]
@juansanchezpena Apologies for not getting back. I'm currently releasing 22.0.0rc4. I've tested on your data and am able to run FreeSurfer through fMRIPrep. This issue should be resolved.

**Comment by juansanchezpena** [comment]
Hey Chris
Just got back from vacation, 22.0.0rc4 isnt on git.
Let me know
J
________________________________

**Comment by effigies** [comment]
I forgot to create a GitHub release (just did), but the image is up on Docker and the package is up on PyPI.

**Comment by juansanchezpena** [comment]
Thanks!
Will test and let you know
J
________________________________

**Comment by juansanchezpena** [comment]
After:cat recon-a

Hey Chris
I for the following error:





Working directory: /tmp/work/fmriprep_22_0_wf/single_subject_patensasc0085_wf/anat_preproc_wf/surface_recon_wf/autorecon_resume_wf/parcstats/mapflow/_parcstats1


Node inputs:


FLAIR_file = <undefined>

T1_files = <undefined>

T2_file = <undefined>

args = <undefined>

big_ventricles = <undefined>

brainstem = <undefined>

directive = autorecon-hemi

environ = {}

expert = <undefined>

flags = ['-nohyporelabel']

hemi = rh

hippocampal_subfields_T1 = <undefined>

hippocampal_subfields_T2 = <undefined>

hires = <undefined>

mprage = <undefined>

mri_aparc2aseg = <undefined>

mri_ca_label = <undefined>

mri_ca_normalize = <undefined>

mri_ca_register = <undefined>

mri_edit_wm_with_aseg = <undefined>

mri_em_register = <undefined>

mri_fill = <undefined>

mri_mask = <undefined>

mri_normalize = <undefined>

mri_pretess = <undefined>

mri_remove_neck = <undefined>

mri_segment = <undefined>

mri_segstats = <undefined>

mri_tessellate = <undefined>

mri_watershed = <undefined>

mris_anatomical_stats = <undefined>

mris_ca_label = <undefined>

mris_fix_topology = <undefined>

mris_inflate = <undefined>

mris_make_surfaces = <undefined>

mris_register = <undefined>

mris_smooth = <undefined>

mris_sphere = <undefined>

mris_surf2vol = <undefined>

mrisp_paint = <undefined>

openmp = 8

parallel = <undefined>

steps = <undefined>

subject_id = sub-patensasc0085

subjects_dir = /output/freesurfer

talairach = <undefined>

use_FLAIR = <undefined>

use_T2 = <undefined>

xopts = <undefined>


Traceback (most recent call last):

  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node

    result["result"] = node.run(updatehash=updatehash)

  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 524, in run

    result = self._run_interface(execute=True)

  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 642, in _run_interface

    return self._run_command(execute)

  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 750, in _run_command

    raise NodeExecutionError(

nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node _parcstats1.


RuntimeError: subprocess exited with code 1.



after

cat recon-all-rh.log


ERROR: Label FG1.mpm.vpnl does not exist in SUBJECTS_DIR fsaverage!

       The fsaverage link probably points to an older freesurfer version

-------------------------------------------------------------------------

Linux 63de26bad2ba 3.10.0-1160.66.1.el7.x86_64 #1 SMP Wed May 18 16:02:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux


recon-all -s sub-patensasc0085 exited with ERRORS at Mon Jul 25 23:47:29 UTC 2022


To report a problem, see http://surfer.nmr.mgh.harvard.edu/fswiki/BugReporting

________________________________

**Comment by effigies** [comment]
Did you delete your fsaverage directory before starting? Or is it linking to a FS6 fsaverage?

**Comment by juansanchezpena** [comment]
I did not delete it
I will do that and re run

________________________________

**Comment by aiims-himanshu** [comment]
i just came across the same error reinstalled to latest one and did clean run have same error

[crash-20220727-095853-root-autorecon1-e0edee1d-98d9-4742-a9f5-4cdb068b4af4.txt](https://github.com/nipreps/fmriprep/files/9197861/crash-20220727-095853-root-autorecon1-e0edee1d-98d9-4742-a9f5-4cdb068b4af4.txt)

**Comment by effigies** [comment]
@aiims-himanshu You're speaking of 22.0.0rc4? And what is the output of the FreeSurfer log?

**Comment by aiims-himanshu** [comment]
I did the run 1 using this opensource data https://openneuro.org/datasets/ds000221/versions/1.0.0 the crash file above have corresponding reconlog1 file 
[recon-all1.log](https://github.com/nipreps/fmriprep/files/9206689/recon-all1.log)

i did a rerun with my own acquired data this time it ran correctly, not sure how but i received an error log in between processing step but output file was generated correctly. 
reconall log for the same attached. 
[recon-all.log](https://github.com/nipreps/fmriprep/files/9206751/recon-all.log)

**Comment by aiims-himanshu** [comment]
Running one more session again (clean) to check for the error log in acquired data and opensource both to check for discrepancy

**Comment by aiims-himanshu** [comment]
> Running one more session again (clean) to check for the error log in acquired data and opensource both to check for discrepancy

It is running in data that i have acquired. 

however the auto recon error is persistent when using the opensource study mentioned above . with following log 
[Node] Error on "fmriprep_22_0_wf.single_subject_01_wf.anat_preproc_wf.surface_recon_wf.autorecon1" (/w/fmriprep_22_0_wf/single_subject_01_wf/anat_preproc_wf/surface_recon_wf/autorecon1)
220728-11:14:56,278 nipype.workflow ERROR:
	 Node autorecon1 failed to run on host e723c29a901e.
220728-11:14:56,278 nipype.workflow ERROR:
	 Saving crash info to /out/sub-01/log/20220728-110550_efd81f90-a927-4b4e-ac78-1bd5e8094768/crash-20220728-111456-root-autorecon1-f228a23f-6818-4741-8b0c-2ec30735e2c9.txt
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 524, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 642, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 750, in _run_command
    raise NodeExecutionError(
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node autorecon1.

RuntimeError: subprocess exited with code 1.

**Comment by effigies** [comment]
ds000221 anatomicals, I think, are MP2RAGE: https://openneuro.org/datasets/ds000221/versions/1.0.0/file-display/sub-010001:ses-02:anat:sub-010001_ses-02_acq-mp2rage_T1w.nii.gz

Handling these is an open problem (https://github.com/nipreps/smriprep/issues/18) and out-of-scope for this issue.

**Comment by juansanchezpena** [comment]
Thanks Chris
Data went through

________________________________
